### ISSUE TITLE
mail confirmation has invalid FROM and DOMAIN (it's hardcoded)

### ISSUE THREAD

**Issue by ghost** [ging/fiware-idm-deprecated#27]
Hi,

I configured successfully the email following this guide
http://fiware-idm.readthedocs.org/en/latest/setup.html#email-configuration

But the source of the email that I receive says:

```
(Authenticated sender: no-reply@organization.org)
From: no-reply@account.lab.fiware.org
```

The activation link has bad domain:

```
https://account.lab.fiware.org/activate/?activation_key=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX&user=user3213
```

if I change to this:

```
https:/organization.org/activate/?activation_key=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX&user=user3213
```

and it works, so where I can change these variables?
I think they are not in /opt/idm/horizon/openstack_dashboard/local/local_settings.py

Neither is explained here: 
http://fiware-idm.readthedocs.org/en/latest/admin_guide.html

Thanks

**Comment by federicofdez** [comment]
Hi,

You are right. This is a known issue, and we solved it partially a couple weeks ago (ging/horizon@69c6a2ea240e3bad9ad97bf079da6d327d847b39). The invalid `from:` field has been fixed in ging/horizon@422a07eb45f57e5393eb70177f0d7b6cc0ba51c7.

Notice that this hasn't been merged yet, but we're planning to include it in the next KeyRock release, which may come out within the next few weeks.

Thanks for reporting the issue.

**Comment by ghost** [comment]
As I understand next version fix this easily. I suppose as another variable in django config of horizon.

But now, for the current version, perhaps that variable is hardcoded in another place, where? Thanks

**Comment by garcianavalon** [comment]
Yeah, the from_email was hardcoded.

Now we changed it to use the django setting DEFAULT_FROM_EMAIL, v4.4.2 should be ready for next week.

If you want to fix it yourself, you can cherry pick the commits from https://github.com/ging/horizon/commits/development, do a merge yourself or replace the hardcoded values to your harcoded email. If you want to see where are all the hardcoded values, take a look at this commit https://github.com/ging/horizon/commit/6ff503d2aa8929ac86ad726bcd016d2b155f7121

**Comment by federicofdez** [comment]
I'll close this issue now, but feel free to reopen it if you have further doubts.

Thanks again.

**Comment by ghost** [comment]
Why a bug is closed if is not solved?

**Comment by garcianavalon** [comment]
Fixed it in https://github.com/ging/horizon/commit/6ff503d2aa8929ac86ad726bcd016d2b155f7121

**Comment by ghost** [comment]
Nice that is solved, sorry if I didn't understand and/or the mood.

Let me share with all of you other parts that are hardcoded and where (if this is useful):

I used this program
`apt-get install silversearcher-ag`

email replacement (this is theoretically solved with ging/horizon@6ff503d):
`ag -l -u --nocolor no-reply@account.lab.fiware.org | xargs perl -pi -e "s/account.lab.fiware.org/${IDM_MAIL}/g"`

horizon web access replacement:
`ag -l -u --nocolor https://account.lab.fiware.org | xargs perl -pi -e "s/\Qhttps://account.lab.fiware.org\E/${IDM_URL}/g"`

hostname replacement:
`ag -l -u --nocolor account.lab.fiware.org | xargs perl -pi -e "s/account.lab.fiware.org/${IDM_HOST}/g"
}`
